src/types/string.rs

```diff
@@ -762,4 +844,26 @@ mod tests {
             assert_eq!(py_string.to_string_lossy(py), "🐈 Hello ���World");
         })
     }
+
+    #[test]
+    fn test_comparisons() {
+        Python::with_gil(|py| {
+            let s = "hello, world";
+            let py_string = PyString::new_bound(py, s);
+
+            assert_eq!(py_string, "hello, world");
+
+            assert_eq!(py_string, s);
+            assert_eq!(&py_string, s);
+            assert_eq!(s, py_string);
+            assert_eq!(s, &py_string);
```

Comment:
Do we want to also add the variants with `*s` to also test the implementations targeting `str` and not `&str`?

Reply:
Good catch, done 👍